test/Electrical/dc_motor.jl

```diff
@@ -0,0 +1,140 @@
+using ModelingToolkit, OrdinaryDiffEq
+using Plots, Test
+
+@parameters t
+
+@testset "DC motor" begin
+    @named ground = Ground()
+    @named source = SignalVoltage()
+    @named voltage_step = Blocks.Step(height=1, start_time=0)
+    @named R1 = Resistor(R=0.5)
+    @named L1 = Inductor(L=4.5e-3)
+    @named emf = EMF(k=0.5)
+    @named fixed = Fixed()
+    @named load = Torque(use_support=false)
+    @named load_step = Blocks.Step(height=-0.3, start_time=3)
+    @named inertia = Inertia(J=0.02)
+    @named friction = RotationalFriction(f=0.01, tau_c=0.2, w_brk=0.1, tau_brk=0.25)
+
+    connections = [
+        connect(fixed.flange, emf.support, friction.flange_b)
+        connect(emf.flange, friction.flange_a, inertia.flange_a)
+        connect(inertia.flange_b, load.flange)
+        connect(load_step.output, load.tau)
+        connect(voltage_step.output, source.input)
+        connect(source.p, R1.p)
+        connect(R1.n, L1.p)
+        connect(L1.n, emf.p)
+        connect(emf.n, source.n, ground.g)
+    ]
+
+    @named model = ODESystem(connections, t, systems=[ground, voltage_step, source, R1, L1, emf, fixed, load, load_step, inertia, friction])
+    sys = structural_simplify(model)
+    prob = ODAEProblem(sys, Pair[], (0, 6.0))
+    sol = solve(prob, Rodas4())
+
+    @test -0.5 .* sol[emf.i] == sol[emf.flange.tau]
+    @test sol[emf.v] == 0.5 .* sol[emf.w]
+
+    # p1 = Plots.plot(sol, vars=[inertia.w], ylabel="Angular Vel. in rad/s", label="")
+    # p2 = Plots.plot(sol, vars=[emf.i], ylabel="Current in A", label="")
+    # Plots.plot(p1, p2, layout=(2,1))
+    # Plots.savefig("dc_motor.png")
+end
+
+@testset "DC motor with speed sensor" begin
+    @named ground = Ground()
+    @named source = SignalVoltage()
+    @named voltage_step = Blocks.Step(height=1, start_time=0)
+    @named R1 = Resistor(R=0.5)
+    @named L1 = Inductor(L=4.5e-3)
+    @named emf = EMF(k=0.5)
+    @named fixed = Fixed()
+    @named load = Torque(use_support=false)
+    @named load_step = Blocks.Step(height=-0.3, start_time=3)
+    @named inertia = Inertia(J=0.02)
+    @named friction = RotationalFriction(f=0.01, tau_c=0.2, w_brk=0.1, tau_brk=0.25)
+    @named speed_sensor = SpeedSensor()
+
+    connections = [
+        connect(fixed.flange, emf.support, friction.flange_b)
+        connect(emf.flange, friction.flange_a, inertia.flange_a)
+        connect(inertia.flange_b, load.flange)
+        connect(inertia.flange_b, speed_sensor.flange)
+        connect(load_step.output, load.tau)
+        connect(voltage_step.output, source.input)
+        connect(source.p, R1.p)
+        connect(R1.n, L1.p)
+        connect(L1.n, emf.p)
+        connect(emf.n, source.n, ground.g)
+    ]
+
+    @named model = ODESystem(connections, t, systems=[ground, voltage_step, source, R1, L1, emf, fixed, load, load_step, inertia, friction, speed_sensor])
+    sys = structural_simplify(model)
+    @test_broken prob = ODAEProblem(sys, Pair[], (0, 6.0)) # KeyError: key 17 not found
```

Comment:
Again the key error

Reply:
@YingboMa is this another case where I should use a `DAEProblem`?

Reply:
Try to use the latest MTK. It should give you a more helpful error.